### ISSUE TITLE
Spikes in training loss

### ISSUE THREAD

**Issue by bilgineroral** [SihanXU/nepa#7]
Hello, thanks for open sourcing the code and really appreciate the work!

I did some modifications on the architecture and was running pretraining on a smaller custom dataset. However, although I'm shuffling the data each epoch there are sudden jumps every few epochs in the training loss curves. Did you observe a similar behavior in your training? I might understand if this is expected though, since the targets are also moving with the patch embedding weights -- but just wanted to confirm.

Thank you!

**Comment by SihanXU** [comment]
We did not observe notable loss spikes during pretraining. For reference, Figure 7 in the paper shows our pretraining loss curves. Could you share a bit more detail about your setup and the loss curve around the spikes?

**Comment by bilgineroral** [comment]
I think I figured it out, I had added a final head on the output interface (e.g. `nn.Linear(config.hidden_size, config.hidden_size)`), but it turns out it didn't make things smoother. Removed it and now it looks more or less like Figure 7.

Again, appreciate the work!
